server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocatorTests.java

```diff
@@ -485,4 +530,11 @@ public ShardAllocationDecision decideShardAllocation(ShardRouting shard, Routing
             }
         };
     }
+
+    private static void rerouteAndWait(AllocationService service, ClusterState clusterState, String reason) throws InterruptedException {
+        var latch = new CountDownLatch(1);
+        var listener = ActionListener.<Void>running(latch::countDown);
+        service.reroute(clusterState, reason, listener);
+        assertTrue("Listener was not called in time", latch.await(10, TimeUnit.SECONDS));
```

Comment:
Think I'd prefer this (which avoids the checked exception too)

```suggestion
        PlainActionFuture.<Void,RuntimeException>get(fut -> service.reroute(clusterState, reason, fut), 10, TimeUnit.SECONDS);
```